docs/references/security/rust-canister-development-security-best-practices.md

```diff
@@ -183,43 +183,179 @@ A canister could be rendered unusable so it could never be upgraded again e.g. d
 
 ## Inter-Canister Calls and Rollbacks
 
-### Don’t panic after await and don’t lock shared resources across await boundaries
+### Message Execution Basics
+
+To understand the issues around async inter-canister calls, one needs to understand a few properties about message execution. This is also explained in the [community conversation on Security Best Practices](https://www.youtube.com/watch?v=PneRzDmf_Xw&list=PLuhDt1vhGcrez-f3I0_hvbwGZHZzkZ7Ng&index=2&t=4s). 
+
+We fist provide a few definitions. A _call_ is a canister's implementation of either an [update](https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-call) or [query call](https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-query) that it exposes. For example, if the Rust CDK is used, these are usually annotated with `#[query]` or `#[update]`, respectively. A _message_ is a set of consecutive instructions that a subnet executes for a canister. We'll see in the following that a call can be split into several messages if inter-canister calls are made. The following properties are essential: 
+
+**Property 1**: Only a single message is processed at a time in a subnet. So message execution is sequential, and never parallel.
+
+**Property 2**: Each call (query / update) triggers a message. When an inter-canister call is made, the code after the call (the callback, highlighted in blue) is executed as a separate message. 
```

Comment:
I think this isn't entirely correct, though I'm not a 100% as I don't understand all the details around the CDK implementation: if your method issues a call, but it doesn't `.await` on the result, I *think* the code after the call would be executed in the same message. I'm even less sure about Motoko.